code/clue.js

```diff
@@ -1,59 +1,351 @@
+  }, 2000);
 
-// This function will be invoked when you click on the killer card.
-const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+  const killerCard = document.getElementById("front-side-killer");
+  showLoader(killerCard);
+};
+
+const pickWeapon = () => {
+  setTimeout(() => {
+    weaponCard.classList.add("hidden");
+    mystery.weapon = randomSelector(weapons);
+    document.getElementById("weapon-card").style.background =
+      mystery.killer.color;
+
+    document.getElementById("weapon-card").style.background =
+      mystery.weapon.color;
+    document.getElementById("weapon-name").innerHTML = `${mystery.weapon.name}`;
+    document.getElementById("weapon-image").classList.remove("hidden");
+    document.getElementById("third-weapon").classList.remove("hidden");
+    document.getElementById("weapon-container").classList.remove("hidden");
+    document.getElementById("weapon-image").src = mystery.weapon.image;
+    document.getElementById(
+      "weapon-weight"
+    ).innerHTML = `${mystery.weapon.weight} lbs`;
+    document.getElementById(
+      "weapon-description"
+    ).innerHTML = `${mystery.weapon.description}`;
+  }, 2000);
+
+  const weaponCard = document.getElementById("front-side-weapon");
+  showLoader(weaponCard);
+};
+
+const pickRoom = () => {
+  const roomCard = document.getElementById("front-side-room");
+  showLoader(roomCard);
+  setTimeout(() => {
+    roomCard.classList.add("hidden");
+    mystery.room = randomSelector(rooms);
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
-}
+    document.getElementById("room-card").style.background =
+      mystery.killer.color;
+    document.getElementById("room-name").innerHTML = `${mystery.room.name}`;
+    document.getElementById("room-container").classList.remove("hidden");
+    document.getElementById("third-room").classList.remove("hidden");
+    document.getElementById("room-image").classList.remove("hidden");
+    document.getElementById("room-image").src = mystery.room.image;
+  }, 2000);
+  
+};
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+const revealMystery = () => {
+  document.getElementById("message-box").classList.remove("hidden");
+  document.getElementById("instructions").classList.add("hidden");
 
-// STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
-// 'The murder was committed by Jacob Green, in the living room with a rope.'
+  if (mystery.killer === "" || mystery.weapon === "" || mystery.room === "") {
```

Comment:
Shorter version of that can be `if (!mystery.killer || !mystery.weapon || !mystery.room) { ... }`